examples/hello-flake/flake.nix

```diff
@@ -0,0 +1,27 @@
+{
+  description = "An example devShell with flakes";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-utils.url = "github:numtide/flake-utils";
```

Comment:
It looks like there is a tendency to not use `flake-utils` in the new projects (either by replacing it with `flake-parts` or its manually written equivalent), so I guess it would be better to do the same here.